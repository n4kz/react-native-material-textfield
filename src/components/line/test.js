import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Line from '.';

/* eslint-env jest */

[{ type: 'solid', color: 'black', width: 1 }, { type: 'dotted', color: 'grey', width: 1 }]
  .forEach(({ type, color, width }) => {
    it(`renders ${type} ${color} line`, () => {
      let line = renderer
        .create(<Line type={type} color={color} width={width} />)
        .toJSON();

      expect(line)
        .toMatchSnapshot();
    });
  });
