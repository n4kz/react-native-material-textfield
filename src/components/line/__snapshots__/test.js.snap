// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dotted grey line 1`] = `
<View
  pointerEvents="none"
  style={
    Array [
      Object {
        "position": "absolute",
      },
      Object {
        "borderColor": "grey",
        "borderStyle": "dotted",
        "borderWidth": 1,
        "bottom": 0,
        "left": -1.5,
        "right": -1.5,
        "top": -2,
      },
    ]
  }
/>
`;

exports[`renders solid black line 1`] = `
<View
  pointerEvents="none"
  style={
    Array [
      Object {
        "position": "absolute",
      },
      Object {
        "borderColor": "black",
        "borderStyle": "solid",
        "borderWidth": 1,
        "bottom": 0,
        "left": -1.5,
        "right": -1.5,
        "top": -2,
      },
    ]
  }
/>
`;
